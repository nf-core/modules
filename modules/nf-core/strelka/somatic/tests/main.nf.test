nextflow_process {

    name "Test Process STRELKA_SOMATIC"
    script "../main.nf"
    process "STRELKA_SOMATIC"

    tag "modules"
    tag "modules_nfcore"
    tag "strelka"
    tag "strelka/somatic"

    test("human - cram") {

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test2.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test2.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true),
                            [],[],
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz.tbi', checkIfExists: true)
                            ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    path(process.out.vcf_indels.get(0).get(1)).vcf.variantsMD5,
                    process.out.vcf_indels_tbi.collect { file(it[1]).getName() },
                    path(process.out.vcf_snvs.get(0).get(1)).vcf.variantsMD5,
                    process.out.vcf_snvs_tbi.collect { file(it[1]).getName() },
                    process.out.versions
                    ).match() }
            )
        }

    }

    test("human - bam") {

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam.bai', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test2.paired_end.recalibrated.sorted.bam', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test2.paired_end.recalibrated.sorted.bam.bai', checkIfExists: true),
                            [],[],
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz.tbi', checkIfExists: true)
                            ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    path(process.out.vcf_indels.get(0).get(1)).vcf.variantsMD5,
                    process.out.vcf_indels_tbi.collect { file(it[1]).getName() },
                    path(process.out.vcf_snvs.get(0).get(1)).vcf.variantsMD5,
                    process.out.vcf_snvs_tbi.collect { file(it[1]).getName() },
                    process.out.versions
                    ).match() }
            )
        }

    }

    test("human - cram - best-practice") {

        config "./nextflow.config"

        when {
            params {
                module_args = '--exome'
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test2.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test2.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome_21.somatic_sv.vcf.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome_21.somatic_sv.vcf.gz.tbi', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz.tbi', checkIfExists: true)
                            ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    path(process.out.vcf_indels.get(0).get(1)).vcf.variantsMD5,
                    process.out.vcf_indels_tbi.collect { file(it[1]).getName() },
                    path(process.out.vcf_snvs.get(0).get(1)).vcf.variantsMD5,
                    process.out.vcf_snvs_tbi.collect { file(it[1]).getName() },
                    process.out.versions
                    ).match() }
            )
        }

    }

    test("human - cram - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test2.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test2.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true),
                            [],[],
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed.gz.tbi', checkIfExists: true)
                            ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
