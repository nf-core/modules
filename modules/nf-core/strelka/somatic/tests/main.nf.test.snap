{
    "human - cram - best-practice": {
        "content": [
            "a500521b4aa64e8f247d57fcb3195b39",
            [
                "test.somatic_indels.vcf.gz.tbi"
            ],
            "17d4e21e50940b4e75cca144af09ce66",
            [
                "test.somatic_snvs.vcf.gz.tbi"
            ],
            [
                "versions.yml:md5,97860fb8f9cda9494819160f7cf6e35b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-08T14:18:48.400058"
    },
    "human - cram - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_indels.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_indels.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_snvs.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_snvs.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,97860fb8f9cda9494819160f7cf6e35b"
                ],
                "vcf_indels": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_indels.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "vcf_indels_tbi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_indels.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf_snvs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_snvs.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "vcf_snvs_tbi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.somatic_snvs.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,97860fb8f9cda9494819160f7cf6e35b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T15:08:42.361540303"
    },
    "human - cram": {
        "content": [
            "6b8a01b52186fcaf3ec74e5bf1b5426",
            [
                "test.somatic_indels.vcf.gz.tbi"
            ],
            "866236c526138e4e08fdc2fa87b08417",
            [
                "test.somatic_snvs.vcf.gz.tbi"
            ],
            [
                "versions.yml:md5,97860fb8f9cda9494819160f7cf6e35b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-08T14:17:42.760334"
    },
    "human - bam": {
        "content": [
            "6b8a01b52186fcaf3ec74e5bf1b5426",
            [
                "test.somatic_indels.vcf.gz.tbi"
            ],
            "866236c526138e4e08fdc2fa87b08417",
            [
                "test.somatic_snvs.vcf.gz.tbi"
            ],
            [
                "versions.yml:md5,97860fb8f9cda9494819160f7cf6e35b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-08T14:48:34.656324"
    }
}